### ISSUE TITLE
Handle returned data containing arrays of length 1 or 0

### ISSUE THREAD

**Issue by digorithm** [FuelLabs/fuels-rs#162]
Currently blocked by https://github.com/FuelLabs/sway/issues/968. The full context is in it. 

In the SDK, this simply means extending:

```rust
pub fn bigger_than_word(&self) -> bool {
    match &*self {
        // Bits256 Always bigger than one `WORD`.
        Self::B256 => true,
        // Strings are bigger than one `WORD` when its size > 8.
        Self::String(size) => size > &8,
        Self::Struct(params) => match params.len() {
            // If only one component in this struct
            // check if this element itself is bigger than a `WORD`.
            1 => params[0].bigger_than_word(),
            _ => true,
        },
        // Enums are always in `ReturnData`.
        Self::Enum(_params) => true,
        // Arrays seem to always be inside `ReturnData`.
        Self::Array(_params, _l) => true,
        // The other primitive types are inside `Return`,
        // thus smaller than one `WORD`.
        _ => false,
    }
}
```

To check for the length of the array and, depending on the logic decided in https://github.com/FuelLabs/sway/issues/968, either grab the returned data from the `Return` receipt or the `ReturnData` receipt.

**Comment by digorithm** [comment]
Closing in favor of this https://github.com/FuelLabs/fuels-rs/issues/233.
